post.md

```diff
@@ -0,0 +1,166 @@
+# Using Zod for Form Validation with React Server Actions in Next.js
+
+In this blog post, we will explore how to use [Zod](https://zod.dev/), a declarative JavaScript validation library, for form validation in a [Next.js](https://nextjs.org/) application. We will also discuss the importance of server-side validation and how to handle validation errors returned from the server.
+
+## Sample Application
+
+This article assumes you're acquainted with setting up and running Next.js applications.
+
+To view the code referenced in this article, checkout the [repo](https://github.com/CodingZeal/nextjs-server-action-validation).
+
+This article and accompanying example application utilize Next.js version 14.x configured with TailwindCSS and TypeScript.
+
+## The Importance of Server-Side Validation
+
+While client-side validation provides immediate feedback and improves user experience, server-side validation is a must for any application. It acts as a second line of defense against invalid or malicious data. Even if you augment your application with client-side validation, server-side validation is necessary to ensure data integrity and security.
+
+## Overview of Zod Schema for Validation
+
+Zod allows us to define a validation schema for our form data. This schema is a declarative way of specifying the validation rules for each field. For example, to mark a field as required, we can use the `min(1)` method, which specifies that the field must have a minimum length of 1.
+
+Here's an example of a Zod schema definition:
+
+```ts
+import { z } from "zod";
+
+const schema = z.object({
+  name: z.string().min(1, "Name cannot be blank"),
+  email: z.string().email("Invalid email address"),
+});
+```
+
+## Handling Server-Side Validation Errors
+
+When we submit our form, the server-side action validates the form data against the Zod schema. If the validation fails, the server returns an array of errors (`ZodIssue[]`). Each error object has the following structure:
```

Comment:
I'd suggest a transition sentence here. Something like, "Using Zod, let's tease out a validation flow that integrates with Next.js. When we submit our form ...."

Reply:
Going to stick with my bland copy for this paragraph